File: backend/src/main/java/com/minetec/backend/dto/form/VehicleTypeCreateUpdateForm.java
package com.minetec.backend.dto.form;

import lombok.Data;

import java.util.List;
import java.util.UUID;

@Data
public class VehicleTypeCreateUpdateForm {
    private String vehicleTypeDesc;
    private List<VehicleTypeAttributeCreateUpdateForm> vehicleTypeAttributeCreateUpdateForms;
    private UUID uuid;
}

File: frontend/src/app/models/vehicle/vehicle-type-create-update-form.ts
import {VehicleTypeAttributeDto} from '../map/VehicleTypeAttributeDto';

export class VehicleTypeCreateUpdateForm {
    public uuid: string;
    public name: string;
    public vehicleTypeDesc: string;
    public vehicleTypeAttributes: VehicleTypeAttributeDto[];
    public msgId;
    public text;
    public type;
    public vehicleTypeAttributeCreateUpdateForms: VehicleTypeAttributeDto[];
}
